### PR TITLE
[feature](nereids) binding slot in order by that not show in project

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

1. binding slot in order by that not show in project, such as:
```sql
SELECT c1 FROM t WHERE c2 > 0 ORDER BY c3
```

2. not check unbound when bind slot reference. Instead, do it in analysis check.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ResolveSortProject.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * resolve sort to add slot reference in sort into it's child - project if necessary. such as:
+ * <p><code>SELECT c1 FROM t WHERE c2 > 0 ORDER BY c3<code />
+ */
+public class ResolveSortProject extends OneAnalysisRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalSort(logicalProject()).when(sort -> sort.getExpressions().stream()
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ResolveSortProject.java:42]:
```suggestion
        return logicalSort(logicalProject()).when(Plan::canBind).when(sort -> sort.getExpressions().stream()
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/AnalyzeRulesJob.java [hunk 2]
```diff
@@ -55,7 +56,8 @@ public AnalyzeRulesJob(CascadesContext cascadesContext, Optional<Scope> scope) {
                         new ProjectToGlobalAggregate()
                 )),
                 topDownBatch(ImmutableList.of(
-                        new ResolveAggregateFunctions()
+                        new ResolveAggregateFunctions(),
+                        new ResolveSortProject()
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/AnalyzeRulesJob.java:60]: Rename to BindXxx?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/AnalyzeRulesJob.java:60]: it is not a bind action. actually, this rule add the projections need by LogicalSort to it's child Project's projections and add another Project on the top of itself to ensure the result is correct

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.79 seconds
 load time: 446 seconds
 storage size: 17178161714 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221108101353_clickbench_pr_41956.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/AnalyzeRulesJob.java (hunk 2 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/AnalyzeRulesJob.java:60]: FillupMissingSlots

**Review by 924060929 — APPROVED** [review]

**Pull request merged**